Deploy-To-ARM Demo (#18)
Azure-Samples/media-services-video-indexer: ARM-Samples/Create-Video-Analyzer/README.md

# Overview

In this Quick-Start you will create an Azure Analyzer for Media (a.k.a "AVAM") account by using Arm Template (PREVIEW)

> The resource will be deployed to your subscription and will create the 'AVAM' resource based on parametrers define in the avam.template file.

Reviewer: ```suggestion
The resource will be deployed to your subscription and will create the Video Analyzer for Media resource based on parameters defined in the avam.template file.

```

> The resource will be deployed to your subscription and will create the 'AVAM' resource based on parametrers presented on the avam.template file.

Reviewer 2: and maybe add the .json extension to the file name here

> **Note:**

> > this sample is *not* for migrating an existing AVAM account to an ARM-Based AVAM account.

Reviewer: ```suggestion
> this sample is *not* for connecting an existing Video Analyzer for Media classic account to an ARM-Based Video Analyzer for Media account.

```

> > For a full documentation on AVAM API, visit the [AVAM Developer Portal](https://aka.ms/avam-dev-portal) page.

Reviewer: ```suggestion
> For full documentation on Video Analyzer for Media API, visit the [Developer Portal](https://aka.ms/avam-dev-portal) page.

```

> > The current API Version is "2021-08-16-preview". Check this Repo from time to time to get updates on new API Versions.

Reviewer 2: We already have 2021-09-01-preview up and running

Author: is there something changed ? This is why I've asked to see an Template sample.

Reviewer 2: yes things changed with the backend of ARM, RPaaS and AVAM between the versions. The template schema is the same, but the request must come in with the new version for it to work properly in the backend.

## Prerequisites

* An Azure Media Service (AMS) account. You can get one for free through the [Create AMS Account](https://docs.microsoft.com/en-us/azure/media-services/latest/account-create-how-to).

## Deploy the sample

----

### Option 1 : Click the "Deploy To Azure Button", and fill in the missing parameters

[![Deploy to Azure](https://aka.ms/deploytoazurebutton)](https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure-Samples%2Fmedia-services-video-indexer%2master%2FARM-Samples%2FCreate-Video-Analyzer%2Favam.template.json)  

----

### Option 2 : Deploy using Power Shell Script

1. Open The [Template File](avam.template.json) file and inspect its content.
2. Fill in the required parameters (see below)
3. Run the Following Power Shell commands:

> * create new Resource group on the same location as your Azure Media Service account, using the [New-AzResourceGroup](https://docs.microsoft.com/en-us/powershell/module/az.resources/new-azresourcegroup) cmdlet.

Reviewer: ```suggestion
* Create a new Resource group on the same location as your Azure Media Service account, using the [New-AzResourceGroup](https://docs.microsoft.com/en-us/powershell/module/az.resources/new-azresourcegroup) cmdlet.

```

> * create new Resource group on the same location as your Azure Media Service account, using the [New-AzResourceGroup](https://docs.microsoft.com/en-us/powershell/module/az.resources/new-azresourcegroup) cmdlet.

Reviewer 2: we recommend having the ams and the avam accounts in the same resource group I think, but as this is generic instructions, maybe leave it as is

```powershell
New-AzResourceGroup -Name myResourceGroup -Location eastus
```

* Deploy the template to the resoruce group using the [New-AzResourceGroupDeployment](https://docs.microsoft.com/en-us/powershell/module/az.resources/new-azresourcegroupdeployment) cmdlet.

```powershell
New-AzResourceGroupDeployment -ResourceGroupName myResourceGroup -TemplateFile ./avam.template.json
```

## Parameters

> ### name

Reviewer: ```suggestion
### Name

```

> * type: string

Reviewer: ```suggestion
* Type: string

```

> * description: Specifies the name of the new Azure Analyzer for Media Resource.

Reviewer: ```suggestion
* Description: Specifies the name of the new Video Analyzer for Media Resource.

```

> * description: Specifies the name of the new Azure Analyzer for Media Resource.

Reviewer: ```suggestion
* Description: Specifies the name of the new Azure Video Analyzer for Media Resource.

```

* required: true

### location

* type: string

> * description: Specifies the Azure location where the AVAM resource should be created.

Reviewer: ```suggestion
* Description: Specifies the Azure location where the AVAM resource should be created.

```

> * required: false

Reviewer: ```suggestion
* Required: false

```

> **Note:**

> > You need to deploy Your Azure Video Analyzer For Media Account in the same region as the Azure Media Services account exists.

Author: yep , I also hope @itay norman can get a technical writer

### mediaServiceAccountResourceId

> * type: string

Reviewer: ```suggestion
* Type: string

```

> * description: The Resource Id of the Azure Media Service Account.

Reviewer: ```suggestion
* Description: The Resource Id of the Azure Media Service Account.

```

* required: true

> ### managedIdentityId

Reviewer: Same here, if possible: Managed Identity ID

* type: string

> * description: The Application Id Of the Managed Identity used to grant access between AVAM resource and the Azure Media Service Account resource

Reviewer 2: it is not the application id, but rather the user assigned managed identity uri (/subscriptions/...resourceGroups/...")

> * description: The Resource Id Of the Managed Identity used to grant access between AVAM resource and the Azure Media Service Account resource

Reviewer: ```suggestion
* Description: The Resource Id Of the Managed Identity used to grant access between AVAM resource and the Azure Media Service Account resource

```

* required: true

### tags

> * type: object

Reviewer: ```suggestion
* Type: object

```

> * description: Array of Objects that represents custom user tags on the AVAM resource

Reviewer: ```suggestion
* Description: Array of Objects that represents custom user tags on the AVAM resource

```

 required: false

### Notes

## Reference Documentation

> If you're new to Video Analyzer, see :

Reviewer: ```suggestion
If you're new to Azure Video Analyzer for Media (formerly Video Indexer), see :

```
